### PR TITLE
[JENKINS-43981] Update to 2.X parent POM

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-43981](https://issues.jenkins-ci.org/browse/JENKINS-43981)

- Update to 2.X parent POM.
- Bump Jenkins and plugins versions.
- Fix Findbugs and tests errors.

@reviewbybees @jglick @kwhetstone @andresrc @bap2000

### CODE REVIEW

File: src/main/resources/org/jenkins_ci/plugins/run_condition/contributed/BatchFileCondition/help.jelly [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<?jelly escape-by-default='true'?>
```
Comment (jglick) [inline src/main/resources/org/jenkins_ci/plugins/run_condition/contributed/BatchFileCondition/help.jelly]: Or just switch to `help.html`.

File: src/test/java/org/jenkins_ci/plugins/run_condition/core/CauseConditionTest.java [hunk 2]
```diff
@@ -220,7 +228,21 @@ private void runtest(List<Cause> causes, RunCondition condition, boolean expecte
         }
         if (causes.size() > 0) {
             // add other causes
-            build.getAction(CauseAction.class).getCauses().addAll(causes);
+            try {
+                build.getAction(CauseAction.class).getCauses().addAll(causes);
+            } catch (UnsupportedOperationException ex) {
```
Comment (jglick) [inline src/test/java/org/jenkins_ci/plugins/run_condition/core/CauseConditionTest.java]: Rather create a new `CauseAction` and `replace` the original.

File: src/test/java/org/jenkins_ci/plugins/run_condition/core/CauseConditionTest.java [hunk 3]
```diff
@@ -241,7 +263,8 @@ private void runtest(List<Cause> causes, RunCondition condition, boolean expecte
     private UserCause createUserCause(String userid) {
         Authentication a = new UsernamePasswordAuthenticationToken(userid, userid);
 
-        a = hudson.getSecurityRealm().getSecurityComponents().manager.authenticate(a);
+        a = j.jenkins.getSecurityRealm().getSecurityComponents().manager.authenticate(a);
```
Comment (jglick) [inline src/test/java/org/jenkins_ci/plugins/run_condition/core/CauseConditionTest.java]: Crazy BTW. Use `User.get('userid').impersonate()` instead. (After ensuring there is a `dummySecurityManager()` at least.)

**Comment by varyvol** [comment]
@jglick I have changed the test code with your suggestions. Thanks!

**Comment by varyvol** [comment]
@imod you merged the latest PRs. Could you review and merge this if appropriate?

**Comment by imod** [comment]
perfect, thanks

**Comment by batmat** [comment]
@imod are you still maintaining this plugin? If so, could you please consider releasing a new version? I'd like to get a fixed version using the latest parent pom to get rid of the warnings (cf. [JENKINS-53285](https://issues.jenkins-ci.org/browse/JENKINS-53285)). Thanks!

**Comment by imod** [comment]
@batmat I was never really maintaining it, but since @bap2000 was not responding to anything anymore, I had to do a release in the past. I need to check if I still have permission to do so...

**Pull request merged**